src/nvim/option_defs.h

```diff
@@ -81,6 +81,12 @@ typedef struct {
   OptValData data;
 } OptVal;
 
+/// Context that an option is being set for.
+typedef struct {
+  win_T *win;
+  buf_T *buf;
+} OptCtx;
```

Comment:
this is probably generalizable / applicable to all kinds of stuff, not only options. similar to `vim._with`. see also `context.c`